### PR TITLE
[SR-3329] URLQueryItems are not getting encoded in to the URL

### PR DESCRIPTION
**Pull request by saiHemak**

### CODE REVIEW

File: CoreFoundation/URL.subproj/CFURLComponents.c [hunk 1]
```diff
@@ -1073,7 +1073,8 @@ CF_EXPORT CFArrayRef _CFURLComponentsCopyQueryItems(CFURLComponentsRef component
                         else {
                             valueString = CFSTR("");
                         }
-                        CFTypeRef keys[] = {CFSTR("name"), CFSTR("value")};
+                        CFStringRef name = CFSTR("name");
+                        CFTypeRef keys[] = {name, CFSTR("value")};
```
Comment (saiHemak) [inline CoreFoundation/URL.subproj/CFURLComponents.c]: To work-around SR-2462 compiler bug.

**Comment by pushkarnk** [comment]
I'm just kicking off a test. Thanks.

@swift-ci please test

**Comment by saiHemak** [comment]
The build  passed successfully in the  local environment .Looking in to the build failure

**Comment by parkera** [comment]
```
Test Case 'TestNSURLComponents.test_createURLWithComponents' started at 08:21:47.819
/home/buildnode/jenkins/workspace/swift-corelibs-foundation-PR-Linux/swift/utils/build-script-impl: line 252:  8657 Segmentation fault      "$@"
/home/buildnode/jenkins/workspace/swift-corelibs-foundation-PR-Linux/swift/utils/build-script: fatal error: command terminated with a non-zero exit status 139, aborting
Build step 'Execute shell' marked build as failure
```

**Comment by pushkarnk** [comment]
We see a similar failure in #713

**Comment by parkera** [comment]
@swift-ci please test

**Comment by pushkarnk** [comment]
A known libkqueue test failure. Triggering the tests again.

@swift-ci please test

**Comment by pushkarnk** [comment]
@swift-ci please test

**Comment by pushkarnk** [comment]
@swift-ci please test

**Comment by naithar** [comment]
Since #713 also crashed the test with the same error, i checked this PR on my 16.04 local environment.  

The issue seems to be in `urlComponents.queryItems` implementation. The exact lines that crashes is: 
```swift
let entryName = oneEntry.object(forKey: "name"._cfObject) as! String
let entryValue = oneEntry.object(forKey: "value"._cfObject) as? String
```

After I changed this lines to:
```swift
let swiftEntry = oneEntry._swiftObject
let entryName = swiftEntry["name"] as! String
let entryValue = swiftEntry["value"] as? String
```
The issue seemed to be fixed and test performed without errors.

**Comment by saiHemak** [comment]
@ naithar : Thanks for debugging ..I was still in process of recreating it as I don't have  Ubuntu 16.0.4 ..

**Comment by pushkarnk** [comment]
@swift-ci please test

**Comment by pushkarnk** [comment]
@swift-ci please test

**Comment by saiHemak** [comment]
@naithar : Thanks again ..It seems the build is passing with your suggested changes .Please let me know if you want to create PR  or shall I go ahead with this PR  including your suggested changes too ..

**Comment by naithar** [comment]
@saiHemak you are welcome :) Feel free to use my suggested changes in this PR, i don't mind.

**Comment by saiHemak** [comment]
@naithar : Thank you :)

**Comment by saiHemak** [comment]
@parkera : Please review  ..

**Comment by parkera** [comment]
Thanks!

**Comment by parkera** [comment]
@swift-ci test and merge

**Pull request merged**